galaxy_ng/tests/integration/utils/iqe_utils.py

```diff
@@ -0,0 +1,193 @@
+        )
+
+
+class TaskWaitingTimeout(Exception):
+    pass
+
+
+class CapturingGalaxyError(Exception):
+    def __init__(self, http_error, message):
+        self.http_error = http_error
+        self.message = message
+
+
+class CompletedProcessError(Exception):
+    pass
+
+
+@lru_cache()
+def get_hub_version(ansible_config):
+    if is_standalone():
+        role = "iqe_admin"
+    else:
+        role = "admin"
+    gc = GalaxyKitClient(ansible_config).gen_authorized_client(role)
+    return gc.get(gc.galaxy_root)["galaxy_ng_version"]
+
+
+def min_hub_version(ansible_config, spec):
+    version = get_hub_version(ansible_config)
+    return Requirement.parse(f"galaxy_ng<{spec}").specifier.contains(version)
+
+
+def max_hub_version(ansible_config, spec):
+    version = get_hub_version(ansible_config)
+    return Requirement.parse(f"galaxy_ng>{spec}").specifier.contains(version)
+
+
+client_cache = {}
+
+
+class GalaxyKitClient:
+    def __init__(self, ansible_config):
+        self.config = ansible_config
+
+    def gen_authorized_client(
+        self,
+        role,
+        container_engine="podman",
+        container_registry=None,
+        *,
+        ignore_cache=False,
+        token=None,
+    ):
+        config = self.config()
+        # role can be either be the name of a user (like `ansible_insights`)
+        # or a dict containing a username and password:
+        # {"username": "autohubtest2", "password": "p@ssword!"}
+        if isinstance(role, dict):
+            cache_key = (role["username"], container_engine, container_registry, token)
+        else:
+            cache_key = (role, container_engine, container_registry, token)
+        ssl_verify = config.get("ssl_verify")
+        if cache_key not in client_cache or ignore_cache:
+            url = config.get("url")
+            if isinstance(role, str):
+                profile_config = self.config(role)
+                user = profile_config.get_profile_data()
+                if profile_config.get("auth_url"):
+                    token = profile_config.get("token")
+                if token is None:
+                    token = get_standalone_token(
+                        user, url, ssl_verify=ssl_verify, ignore_cache=ignore_cache
+                    )
+
+                auth = {
+                    "username": user["username"],
+                    "password": user["password"],
+                    "auth_url": profile_config.get("auth_url"),
+                    "token": token,
+                }
+            else:
+                token = get_standalone_token(
+                    role, url, ignore_cache=ignore_cache, ssl_verify=ssl_verify
+                )  # ignore_cache=True
+                role.update(token=token)
+                auth = role
+
+            container_engine = config.get("container_engine")
+            container_registry = config.get("container_registry")
+
+            g_client = GalaxyClient(
```

Comment:
It looks like this call to galaxykit is causing the pr-check.sh to fail with `KeyError: 'access_token'`